kernel/kernel-defaults/src/test/scala/io/delta/kernel/defaults/ChecksumSuite.scala

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.defaults
+
+import java.util.Optional
+
+import scala.collection.immutable.Seq
+import scala.jdk.CollectionConverters.{asJavaIteratorConverter, mapAsJavaMapConverter}
+
+import io.delta.kernel.{Table, Transaction}
+import io.delta.kernel.expressions.Literal
+import io.delta.kernel.internal.SnapshotImpl
+import io.delta.kernel.internal.checksum.ChecksumReader
+import io.delta.kernel.internal.fs.Path
+import io.delta.kernel.internal.stats.FileSizeHistogram
+import io.delta.kernel.internal.util.Utils.toCloseableIterator
+import io.delta.kernel.utils.CloseableIterable.inMemoryIterable
+import io.delta.kernel.utils.DataFileStatus
+
+/**
+ * Functional e2e test suite for verifying checksum correctness.
+ */
+class ChecksumSuite extends DeltaTableWriteSuiteBase {
+
+  test("Check checksum fields are correct") {
+    withTempDirAndEngine { (tablePath, engine) =>
+      // Create transaction for a new table with test schema
+      val txn = createTxn(engine, tablePath, isNewTable = true, testSchema, Seq.empty)
+      val txnState = txn.getTransactionState(engine)
```

Comment:
we don't need to write each step to create the table. You can just call appendData. Get the list of parquet file status using fsclient.listFrom and compute the expected file count and size.

Reply:
we also test remove file in the PR, which at this moment only works with icebergwritercompatv1, however appendData is not working with column name mapping iiuc.